example/extensions/lib_custom_op/relu_lib.cu

```diff
@@ -180,6 +182,75 @@ REGISTER_OP(my_state_relu)
 .setCreateOpState(createOpStateCPU, "cpu")
 .setCreateOpState(createOpStateGPU, "gpu");
 
+
+
+/* ------------------------ Below is noisy relu operator example ---------------------*/
+
+#include <curand_kernel.h>
+#include <random>
+
+#define NumRandomPerThread 64 // mxnet recommended random numbers generated per thread
+
+__global__ void noisy_relu_gpu_forward(float *out, float *in, int64_t N, void *states, int step) {
+    int tid = blockIdx.x * blockDim.x + threadIdx.x;
+    curandStatePhilox4_32_10_t* global_states = (curandStatePhilox4_32_10_t*)states;
```

Comment:
use static_cast<curandStatePhilox4_32_10_t*>(states)

Reply:
can you add a comment to explain why you're doing this? where do the states get allocated? how can you guarantee the states pointer is pointing to a big enough array to index with tid?

Reply:
Added to comments to explain the logic, and made some type tweaks